stdlib/3/itertools.pyi

```diff
@@ -21,12 +21,14 @@ def repeat(object: _T, times: int) -> Iterator[_T]: ...
 
 if sys.version_info >= (3, 8):
     @overload
-    def accumulate(iterable: Iterable[_T], func: Callable[[_T, _T], _T] = ...) -> Iterator[_T]: ...
+    def accumulate(iterable: Iterable[_T], func: Optional[Callable[[_T, _T], _T]] = ...) -> Iterator[_T]: ...
     @overload
-    def accumulate(iterable: Iterable[_T], func: Callable[[_S, _T], _S], initial: Optional[_S]) -> Iterator[_S]: ...
+    def accumulate(
+        iterable: Iterable[_T], func: Optional[Callable[[_S, _T], _S]] = ..., *, initial: Optional[_S] = ...
+    ) -> Iterator[_S]: ...
```

Comment:
These two overloads seem to overlap. Also, the second overload can't work if `func` is None, because `_S` is then undefined. I would suggest to use `func: None = ...` for the first (which also needs to include `initial`), and `func: Callable[etc]` (without default) for the second overload. I think this should also cover all cases?

Reply:
Thanks, that looks right to me too